internal/app/mymemory/mymemory.go

```diff
@@ -0,0 +1,69 @@
+package mymemory
+
+import (
+	"crypto/rand"
+	"fmt"
+)
+
+type MyMemory struct {
+	myMemory map[string]string
+}
+
+func InitMyMemory(initValue map[string]string) *MyMemory {
+	m := MyMemory{}
+	m.myMemory = initValue
+
+	return &m
+}
+
+func (m *MyMemory) CreateOrGetFromMemory(url []byte) (string, error) {
+	val, ok := m.myMemory[string(url)]
+	// If the key exists
+	if ok {
+		return val, nil
+	}
+	// input into our memory
+	if val, err := createShortURL(url); err == nil {
```

Comment:
```
	if val, err := createShortURL(url); err != nil {
		return "", err
	}

        m.myMemory[string(url)] = val
        return val, nil
```

Reply:
corrected, but now i have 2 similar `return`...
`func (m *MyMemory) CreateOrGetFromMemory(url string) (string, error) {
	val, ok := m.myMemory[string(url)]
	// If the key exists
	if !ok {
		// input into our memory
		val, err := utils.CreateShortURL()
		if err != nil {
			return "", err
		}
		m.myMemory[string(url)] = val
		return val, nil
	}
	return val, nil
}`
I need second return, because local `val` rewrite variable from parent lexical environment... but okaaaay))